File: KITTI_vis_test_seq.m
function KITTI_vis_test_seq
    use_gpu = 1;
   
    %%
    if exist('../+caffe', 'dir')
        addpath('..');
    else
        error('Please run this demo from caffe/matlab/demo');
    end
    
    %% Set caffe mode
    if exist('use_gpu', 'var') && use_gpu
        caffe.set_mode_gpu();
        gpu_id = 0;  % we will use the first gpu in this demo
        caffe.set_device(gpu_id);
    else
        caffe.set_mode_cpu();
    end
    
    
    net_model = 'scripts/deploy_resnet50by2_pool.prototxt';
    net_weights = 'model/train_iter_40000.caffemodel';

 
    phase = 'test';
    if ~exist(net_model, 'file')
        error('Please download CaffeNet from Model Zoo before you run this demo');
    end
    
    net = caffe.Net(net_model, net_weights,phase);
       
    prediction_disp = [];
    for i = 1:4
    im = imread(sprintf('images/%d.png',i));
    im = imresize(im,[160,608]);

    im_data = im(:, :, [3, 2, 1]);  % permute channels from RGB to BGR
    im_data = permute(im_data, [2, 1, 3]);  % flip width and height
    im_data = single(im_data);  % convert from uint8 to single    
    mean_data = repmat(reshape([104 117 123],1,1,3),size(im_data,1),size(im_data,2),1);
    im_data = im_data - mean_data;  % subtract mean_data (already in W x H x C, BGR)
        
    net.blobs('imL').reshape([size(im_data) 1]);
    
    tic;
    scores = net.forward({im_data});
    time_taken(i) = toc;
    
    subplot(2,1,1),imshow(im), title('Left image (Input)');
%     figure(1),subplot(2,1,4),imshow(imR),title('Right image');
%     figure(1),subplot(2,1,3),imshow(permute(scores{1}/255,[2,1,3])),title('back warp');
    A =  net.blobs('h_flow').get_data();
    figure(1),subplot(2,1,2), imagesc(permute(A,[2,1,3]));axis equal;axis off; axis tight;colormap gray,title('Depth');    
    pause(.5)
    
    prediction_disp(:,:,i) = permute(A,[2,1,3]);
    i
    end 
    caffe.reset_all();

    fprintf('Network running at %02g Hz',1/mean(time_taken(2:end)))
    
    %% processing  disparity to depth
    
    % scale the disparities to KITTI resolution
    scale = size(im,2)/size(prediction_disp,2);
    scale = permute(scale,[2 3 1]);
    flowHr =scale*imresize(prediction_disp,[370,1224]); 
    % estimate depth
    predicted_depths = 389.6304./flowHr;%predicted_depths = min(max(389.6304./flowHr,1),50);
    
    % load eigens mask and apply it for evaluation
    mask = logical(imread('images/mask_eigen.png'));
    
    for i = 1:size(predicted_depths,3)
        p = predicted_depths(:,:,i);
        p(~mask(:)) = nan;
        predicted_depths(:,:,i) = p; 
    end
    % evaluate disparity with your own evaluation function
    %A = evaluate_testset_kitti(predicted_depths);
     
1;

File: KITTI_vis_test_seq.py
import matplotlib
matplotlib.use('Agg') # if on remote server
import argparse
import os
import cv2
import caffe
import numpy as np
import matplotlib.pyplot as plt

parser = argparse.ArgumentParser()
parser.add_argument('--eigen_test_files', type=str, default='./eigen_test_files', help='697 eigen split test images')
parser.add_argument('--dump_root', type=str, default='./dumps', help='directory to output disp/depth')
parser.add_argument('--output_disp', action='store_true', help='output disparity')
parser.add_argument('--output_depth', action='store_true', help='otuput depth')

args = parser.parse_args()

caffe.set_mode_gpu()
gpu_id = 0
caffe.set_device(gpu_id)

net_model = "./scripts/deploy_resnet50by2_pool.prototxt"
net_weights = "./model/train_iter_40000.caffemodel"

net = caffe.Net(net_model, net_weights, caffe.TEST)

imgs = os.listdir(args.eigen_test_files)

if not os.path.exists(args.dump_root):
    os.makedirs(args.dump_root)

for i, img in enumerate(imgs):
    im = cv2.imread(os.path.join(args.eigen_test_files, img)).astype(np.float32)
    im = cv2.resize(im, (608, 160))
    im_mean = np.mean(im)
    im -= im_mean
    net.blobs['imL'].reshape(1, im.shape[2], im.shape[0], im.shape[1]) # (1, 3, H, W)
    net.blobs['imL'].data[...] = np.transpose(im[:, :, :, np.newaxis], (3, 2, 0, 1))
    net.forward()
    A = net.blobs['h_flow'].data
    disp = A[0][0] # (H, W)
    file_name = os.path.splitext(img)[0]
    fig = plt.imshow(disp, cmap='plasma')
    plt.axis('off')
    fig.axes.get_xaxis().set_visible(False)
    fig.axes.get_yaxis().set_visible(False)
    plt.savefig(os.path.join(args.dump_root, file_name+'_disp.png'), bbox_inches='tight', pad_inches=0)
